l10n_be_coa_multilang/account_financial_report.py

```diff
@@ -0,0 +1,293 @@
+                if account.company_id.country_id.code in \
+                        self._be_scheme_countries:
+                    acc_code = acc_code or account.code
+                    acc_type = acc_type or account.type
+                    centralized = centralized or account.centralized
+                    be_report_entries = filter(
+                        lambda x: acc_code[
+                            0:len(x['account_group'])] == x['account_group'],
+                        scheme_table)
+                    if len(be_report_entries) > 1:
+                        raise except_orm(
+                            _("Configuration Error !"),
+                            _("Configuration Error in the "
+                              "Belgian Legal Financial Report Scheme."))
+                    be_report_id = be_report_entries and \
+                        be_report_entries[0]['report_id'][0]
+                    for fin_report in account.financial_report_ids:
+                        if fin_report.id in be_report_ids:
+                            if acc_type not in ['view', 'consolidation'] \
+                                    and fin_report.id == be_report_id:
+                                updated = True
+                            elif acc_type == 'view' and centralized \
+                                    and fin_report.id == be_report_id:
+                                updated = True
+                            else:
+                                vals.update({
+                                    'financial_report_ids':
+                                        [(3, fin_report.id)]})
+                                updated = True
+                    if be_report_id and (
+                            acc_type not in ['view', 'consolidation'] or
+                            (acc_type == 'view' and centralized)
+                            ) and not updated:
+                        vals.update(
+                            {'financial_report_ids': [(4, be_report_id)]})
+        return super(account_account, self).write(
+            cr, uid, ids, vals, context=context)
+
+
+class l10n_be_update_be_reportscheme(models.TransientModel):
+    _name = 'l10n_be.update_be_reportscheme'
+    _description = 'Update BNB/NBB financial reports configuration'
+
+    note = fields.Text('Result', readonly=True)
+
+    def _update_be_reportscheme(self):
+        """"
+        This method is executed when installing the module and will
+        create/update the entries in the BNB/NBB legal report scheme.
+        """
+        note = ''
+        upd_ctx = {'update_be_reportscheme': True}
+        scheme_table = self.env['be.legal.financial.reportscheme'].search([])
+        be_report_ids = [x.report_id.id for x in scheme_table]
+        accounts = self.env['account.account'].search(
+            ['|', ('type', '!=', 'view'), '&', ('type', '=', 'view'),
+             ('centralized', '=', True)]
+            ).with_context(upd_ctx)
```

Comment:
It seems this ```with_context``` is missplaced (i.e. the code in ```account.search()``` related to this specific context is never executed). Probably you want to say  ```self.env['account.account'].with_context(upd_ctx).search(...)```